src/test/java/StringManipulationTest.java

```diff
@@ -118,32 +200,51 @@ public void testRestoreString1()
         assertEquals(restoreString, "rat");
     }
 
+    /**
+     * Check whether a simple string can be correctly rearranged into all the same letter (this wasn't part of the spec
+     * but I am making the executive decision to include it).
+     */
     @Test
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("abc");
+        int [] array = new int[]{2,2,2};
+        assertEquals(manipulatedstring.restoreString(array), "ccc");
     }
 
+    /**
+     * Check whether an illegal argument exception is correctly thrown when the number of indices doesn't match
+     * the number of letters in the string.
+     */
     @Test
     public void testRestoreString3()
```

Comment:
Here, you're testing if your method can deal with a mismatch in the number of indices and the number of characters in the string. It's like trying to fit a square peg in a round hole - it's not gonna work, and your method should recognize that. Good on you for checking this!